scripts/dprpwg-gen.py

```diff
@@ -0,0 +1,77 @@
+#! /usr/bin/env python3
+#
+# This script generates a dprpwg_config.h file from dprpwg_config.stubs.h by
+# using pseudo-random numbers. The numbers are generated accourding to the
+# python PEP506 Note. As such, they SHOULD be suitable for cryptographic use.
+
+
+import argparse
+import ctypes
+import secrets
+import sys
+from pathlib import Path
+
+THIS_DIR = Path(__file__).resolve()
+DEFAULT_CONFIG = THIS_DIR.parent.parent / "src" / "dprpwg_config.stub.h"
+
+class Help:
+    TEMPLATE = """Path to the dprpwg_config.sub.h file to be used as a template
+to generate a suitable dprpwg_config.h file"""
+    OUTPUT = """Path to the output header to generate. If not provided, stdout
+will be used"""
+
+def getopts(argv):
+    parser = argparse.ArgumentParser(description='DPRPWG config.h generator')
+    parser.add_argument("--template", "-t", metavar='CONFIG.h',
+                        default=DEFAULT_CONFIG, help=Help.TEMPLATE)
+    parser.add_argument("--output", "-o", metavar='OUTPUT.h',
+                        help=Help.OUTPUT)
+    return parser.parse_args(argv[1:])
+
+def main(argv):
+    args = getopts(argv)
+
+    # Here is the list of all the #define values we shall substitute
+    variables = ("PW_MUL", "PW_SEEK_MUL", "PW_INV_MUL",
+                 "DOM_MUL", "DOM_SEEK_MUL", "DOM_INV_MUL",
+                 "YR_MUL", "YR_SEEK_MUL", "YR_INV_MUL")
+
+    # The values to substitute are 'unsigned int'. So, determine the count of
+    # bits that can be used in the generation of the random values.
+    # We assume 8 bits per byte.
+    bits = ctypes.sizeof(ctypes.c_uint * 8)
+
+    # Make a substitution table that to each variable associates a random
+    # number (as a string)
+    replace_table = {x: str(secrets.randbits(bits)) for x in variables}
```

Comment:
As said in [`dprpwg_config.stub.h`](src/dprpwg_config.stub.h), I think prime numbers are better because of the way these numbers are used by the generation algorithm.

So, if you have a python random prime number generator instead of just random bits, that would be nicer... :)

Reply:
It seems there are no proof that primes are better (to your own telling). Generating random primes seems much more complex, because you have to either test that a given number is a primer number or embed a table of possible 64-bits primes.
There may exist a python module to "generate" random numbers, but I don't believe such a simple tool should request an external dependency.

If you can prove me that primes are better, I'll invest some time in that. Otherwise, I believe generating random numbers will be way better than trying to let the user define its own primes, because I believe the user is not going to find the prime numbers very far from zero (I wouldn't).